tests/test_catch_multi_threaded.nit

```diff
@@ -0,0 +1,43 @@
+# Test the multi-threaded dynamic `do catch` mecanism
+import pthreads
+
+
+# A Thread that do a lot of `do catch`
+class CatchThread
+	super Thread
+
+
+	var x = 200
+	var catched = 0
+
+	redef fun main do
+		do
+			rec_do_catch
+		catch
+		end
+		print "catched {catched} aborts"
+		return null
+	end
+
+	fun rec_do_catch do
+		do
```

Comment:
Can you do this?
~~~
fun foo do
    # ...
catch
    # ...
end
~~~

Reply:
it doesnt work for now, maybe a future PR will allow it